test/integration/load/load_test.go

```diff
@@ -171,9 +178,94 @@ func TestScaleDeployment(t *testing.T) {
 	if testConfig.Cleanup {
 		err = kubernetes.MustDeleteDeployment(ctx, deploymentsClient, deployment)
 		require.NoError(t, err, "error deleteing load deployment")
+		err = kubernetes.WaitForPodsDelete(ctx, clientset, namespace, podLabelSelector)
+		require.NoError(t, err, "error waiting for pods to delete")
 	}
 }
 
+func TestValidCNSStateDuringScaleAndCNSRestart(t *testing.T) {
+	clientset, err := kubernetes.MustGetClientset()
+	require.NoError(t, err)
+
+	config := kubernetes.MustGetRestConfig(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	defer cancel()
+
+	validator, err := validate.CreateValidator(ctx, clientset, config, namespace, testConfig.CNIType, testConfig.RestartCase, testConfig.OSType)
+	require.NoError(t, err)
+
+	err = validator.Validate(ctx)
+	require.NoError(t, err)
+
+	deployment, err := kubernetes.MustParseDeployment(noopDeploymentMap[testConfig.OSType])
+	require.NoError(t, err)
+	deploymentsClient := clientset.AppsV1().Deployments(namespace)
+
+	// Create a deployment
+	err = kubernetes.MustCreateDeployment(ctx, deploymentsClient, deployment)
+	require.NoError(t, err)
+
+	// Scale it up and "skipWait", so CNS restart can happen immediately after scale call is made (while pods are still creating)
+	skipWait := true
+	err = kubernetes.MustScaleDeployment(ctx, deploymentsClient, deployment, clientset, namespace, podLabelSelector, testConfig.ScaleUpReplicas, skipWait)
+	require.NoError(t, err)
+
+	// restart linux CNS (linux, windows)
+	err = kubernetes.RestartCNSDaemonset(ctx, clientset)
+	require.NoError(t, err)
+
+	// wait for pods to settle before checking cns state (otherwise, race between getting pods in creating state, and getting CNS state file)
+	err = kubernetes.WaitForPodDeployment(ctx, clientset, namespace, deployment.Name, podLabelSelector, testConfig.ScaleUpReplicas)
+	require.NoError(t, err)
+
+	// Validate the CNS state
+	err = validator.Validate(ctx)
+	require.NoError(t, err)
+
+	// Scale it down
+	err = kubernetes.MustScaleDeployment(ctx, deploymentsClient, deployment, clientset, namespace, podLabelSelector, testConfig.ScaleDownReplicas, skipWait)
```

Comment:
is there a reason we scale it down ?

Reply:
Yeah there have been bugs before in CNS, when we scale down, that IPs aren't released, and then after restart, those IPs are stuck so we're testing that now